### DISCUSSION TITLE
Override RestAssured port reliably

### DISCUSSION THREAD

**Discussion by languitar** [quarkusio/quarkus#26656] [Community]
Some part of the Quakurs test infrastructure automatically sets the RestAssured configuration for each test. In my case I would like to control this port manually as part of test resource to hook the prism OpenAPI validation proxy into the test harness. Is there any chance to control which port gets configured for RestAssured in a central place?

What I tried is the following:

```kotlin
class PrismProxyResource : QuarkusTestResourceLifecycleManager {

    override fun start(): MutableMap<String, String> {
        Testcontainers.exposeHostPorts(QUARKUS_HTTP_PORT)
        proxy.start()
        RestAssured.port = proxy.getMappedPort(PRISM_PORT)
        return mutableMapOf(
            "quarkus.http.test-port" to QUARKUS_HTTP_PORT.toString(),
        )
    }
```

However, `RestAssuredURLManager` seems to be run for every test. Therefore, my custom port is overwritten.

What I'd like to avoid is having to make modifications to each test class to get this working. Therefore, a central place would be a lot nicer.

**Comment by radcortez** [comment]
Currently, the only way to do it is to set the port in a `beforeEach` method. This could be implemented as a JUnit extension that executes after Quarkus.
